engine/src/it/java/org/pentaho/di/kitchen/KitchenIT.java

```diff
@@ -0,0 +1,320 @@
+
+      } catch ( SecurityException e ) {
+        // All OK / expected: SecurityException is purposely thrown when Kitchen triggers System.exitJVM()
+
+        // get log file contents
+        String logFileContent = getFileContentAsString( logFileRelativePath );
+
+        assertTrue( logFileContent.contains( JOB_EXEC_RESULT_TRUE ) );
+        assertFalse( logFileContent.contains( FAILED_JOB_EXEC_PATTERN ) );
+
+        Result result = Kitchen.getCommandExecutor().getResult();
+        assertNotNull( result );
+        assertEquals( result.getExitStatus(), CommandExecutorCodes.Kitchen.SUCCESS.getCode() );
+
+      } finally {
+        // sanitize
+        File f = new File( logFileFullPath );
+        if ( f != null && f.exists() ) {
+          f.deleteOnExit();
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testFileJobWithParams() throws Exception {
+
+    String param1Name = "p1";
+    String param2Name = "p2";
+
+    String param1Val = UUID.randomUUID().toString();
+    String param2Val = UUID.randomUUID().toString();
+
+    final String EXPECTED_OUTPUT = "Received params " + param1Name + ":" + param1Val + ", " + param2Name + ":" + param2Val;
+
+    long now = System.currentTimeMillis();
+
+    String testKJBRelativePath = EXPECTED_PARAMS_PASSED_ALONG + File.separator + "print_received_params.kjb";
+    String testKJBFullPath = this.getClass().getResource( testKJBRelativePath ).getFile();
+    String logFileRelativePath = testKJBRelativePath + "." + now + ".log";
+    String logFileFullPath = testKJBFullPath + "." + now + ".log";
+
+    try {
+
+      Kitchen.main( new String[] {
+        "/file:" + testKJBFullPath,
+        "/level:Basic",
+        "/logfile:" + logFileFullPath,
+        "/param:" + param1Name + "=" + param1Val,
+        "/param:" + param2Name + "=" + param2Val
+      } );
+
+    } catch ( SecurityException e ) {
+      // All OK / expected: SecurityException is purposely thrown when Pan triggers System.exitJVM()
+
+      // get log file contents
+      String logFileContent = getFileContentAsString( logFileRelativePath );
+
+      assertTrue( logFileContent.contains( EXPECTED_OUTPUT ) );
+      assertTrue( logFileContent.contains( JOB_EXEC_RESULT_TRUE ) );
+      assertFalse( logFileContent.contains( FAILED_JOB_EXEC_PATTERN ) );
+
+      Result result = Kitchen.getCommandExecutor().getResult();
+      assertNotNull( result );
+      assertEquals( result.getExitStatus(), CommandExecutorCodes.Kitchen.SUCCESS.getCode() );
+
+    } finally {
+      // sanitize
+      File f = new File( logFileFullPath );
+      if ( f != null && f.exists() ) {
+        f.deleteOnExit();
+      }
+    }
+  }
+
+  private String getFileContentAsString( String relativePath ) {
+
+    String content = "";
+
+    try {
+      BufferedReader rd = new BufferedReader(
+              new InputStreamReader( this.getClass().getResourceAsStream( relativePath ) ) );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *InputStreamReader*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *InputStreamReader()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).